projects/core/services/test/breakpoint.service.spec.ts

```diff
@@ -0,0 +1,61 @@
+import {TestBed} from '@angular/core/testing';
+import {WINDOW} from '@ng-web-apis/common';
+import {configureTestSuite} from '@taiga-ui/testing';
+
+import {TuiMedia} from '../../interfaces';
+import {TUI_MEDIA} from '../../tokens';
+import {TuiBreakpointService} from '../breakpoint.service';
+
+describe(`TuiBreakpointService`, () => {
+    const mock: HTMLDivElement = document.createElement(`div`);
+    let service: TuiBreakpointService;
+    const mediaMock: TuiMedia = {
+        mobile: 768,
+        desktopSmall: 1024,
+        desktopLarge: 1280,
+    };
+
+    const windowMock: any = {
+        matchMedia: jest
+            .fn()
+            .mockReturnValue({...mock, matches: true, media: `(max-width: 767px)`}),
+        innerWidth: 700,
+    };
+
+    configureTestSuite(() => {
+        TestBed.configureTestingModule({
+            providers: [
+                TuiBreakpointService,
+                {
+                    provide: TUI_MEDIA,
+                    useValue: mediaMock,
+                },
+                {
+                    provide: WINDOW,
+                    useValue: windowMock,
+                },
+            ],
+        });
+    });
+
+    beforeEach(() => {
+        service = TestBed.inject(TuiBreakpointService);
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it(`should create`, () => {
+        expect(service).toBeTruthy();
+    });
+
+    it(`should emit the current breakpoint name when subscribed to`, () => {
```

Comment:
more test please for all `mobile: 768,
        desktopSmall: 1024,
        desktopLarge: 1280,`

Reply:
I could not emit matchMedia events 🥺

Reply:
`breakpoint.cy.ts`

```ts
for (const [index, {width, height}] of [
        {width: 768, height: 900},
        {width: 1024, height: 900},
        {width: 1280, height: 900},
    ].entries()) {
        describe(`${width}x${height}`, () => {
            it(() => {
                cy.viewport(width, height);
                cy.tuiVisit(`/services/breakpoint-service`);

                cy.get(`tui-doc-example[heading="Basic"]`)
                    .findByAutomationId(`tui-doc-example`)
                    .tuiScrollIntoView()
                    .tuiWaitBeforeScreenshot()
                    .matchImageSnapshot(`breakpoint_${width}x${height}`);
            });
        });
}
```

Reply:
is it snapshot testing ? how to run such test locally ?

Reply:
make that file in kit directory inside `demo-integration`

```
$ nx e2e demo
```

Reply:
thank you 😍 it works, i just pushed the test.